model/commitqueue/commit_queue.go

```diff
@@ -47,18 +47,18 @@ func (q *CommitQueue) All() []string {
 	return q.Queue
 }
 
-func (q *CommitQueue) Remove(item string) error {
+func (q *CommitQueue) Remove(item string) (bool, error) {
 	itemIndex := q.findItem(item)
 	if itemIndex < 0 {
-		return errors.New("item not in queue")
+		return false, nil
 	}
 
 	if err := remove(q.ProjectID, item); err != nil {
-		return errors.Wrap(err, "can't remove item")
+		return true, errors.Wrap(err, "can't remove item")
```

Comment:
If I'm reading this correctly, the boolean is not relevant here, since the error is checked first. If that is the case, I think the zero value is the correct value to return, in this case `false`.

Reply:
the motivation here is to differentiate between internal (database) errors and non-existing project/item pairs so we can return the appropriate status code in the API call.
Here, remove will return an error if something goes wrong with the database, but we've already ascertained that it exists (on lines 51-54), so it returns true for found, with the error.

Reply:
Is there a caller that checks if there is an error, and then having found one, also checks the boolean?

Reply:
Yes: the rest route will return a 404 if not found and a 500 if error.